packages/autorest.python/autorest/codegen/serializers/__init__.py

```diff
@@ -532,9 +532,17 @@ def _serialize_and_write_metadata(
             namespace_path / Path("_metadata.json"), metadata_serializer.serialize()
         )
 
+    def _name_space(self) -> str:
+        if self.code_model.namespace.count(".") >= (
+            self.code_model.options["package_name"] or ""
+        ).count("-"):
+            return self.code_model.namespace
+
+        return self.code_model.options["package_name"].replace("-", ".")
+
```

Comment:
In cadl, the namespace usually doesn't contain `Azure.xxx` which is not consistent with package name. And I think it is not convenient to force customers to add it. According to [Azure SDK distribution packages
](https://azure.github.io/azure-sdk/python_design.html#azure-sdk-distribution-packages), namespace shall keep consistent with package name, so we could use package name to generate right package structure which doesn't break old code with swagger.